### PR TITLE
[SMALLFIX] Create journal directory on startup with Embedded

### PR DESCRIPTION
**Pull request by ns1123**
Check if Alluxio is in EMBEDDED mode and if so, create the journal directory if it doesn't exist.

### CODE REVIEW

**Comment by ns1123** [comment]
@calvinjia PTAL

**Comment by ns1123** [comment]
jenkins, test this please

File: bin/alluxio-start.sh [hunk 1]
```diff
@@ -230,6 +230,14 @@ start_logserver() {
 }
 
 start_master() {
+  if [[ `${LAUNCHER} ${BIN}/alluxio getConf alluxio.master.journal.type` == "EMBEDDED" ]]; then
+    JOURNAL_DIR=`${LAUNCHER} ${BIN}/alluxio getConf alluxio.master.journal.folder`
+    if [ -f "${JOURNAL_DIR}" ]; then
+      echo "Journal location ${JOURNAL_DIR} is a file not a directory. Please remove the file before retrying."
+    elif [ ! -e "${JOURNAL_DIR}" ]; then
+      ${LAUNCHER} ${BIN}/alluxio format
```
Comment (calvinjia) [inline bin/alluxio-start.sh]: Should we do this after the `-f` check to prevent double format?

File: bin/alluxio-start.sh [hunk 2]
```diff
@@ -230,6 +230,14 @@ start_logserver() {
 }
 
 start_master() {
+  if [[ `${LAUNCHER} ${BIN}/alluxio getConf alluxio.master.journal.type` == "EMBEDDED" ]]; then
```
Comment (apc999) [inline bin/alluxio-start.sh]: `${LAUNCHER} ${BIN}/alluxio getConf ${ALLUXIO_MASTER_JAVA_OPTS} alluxio.master.journal.type` == "EMBEDDED"

File: bin/alluxio-start.sh [hunk 3]
```diff
@@ -230,6 +230,14 @@ start_logserver() {
...
+    JOURNAL_DIR=`${LAUNCHER} ${BIN}/alluxio getConf alluxio.master.journal.folder`
```
Comment (apc999) [inline bin/alluxio-start.sh]: ditto

Comment (ns1123) [reply bin/alluxio-start.sh]: done

File: bin/alluxio-start.sh (hunk 2 above)
Comment (ns1123) [reply bin/alluxio-start.sh]: done

File: bin/alluxio-start.sh (hunk 1 above)
Comment (ns1123) [reply bin/alluxio-start.sh]: done

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Pull request merged**